packages/mgt-components/src/components/mgt-people-picker/mgt-people-picker.ts

```diff
@@ -984,9 +1003,38 @@ export class MgtPeoplePicker extends MgtTemplatedComponent {
       this.hideFlyout();
       // fire selected people changed event
       this.fireCustomEvent('selectionChanged', this.selectedPeople);
+    } else if (event.code === 'Comma' || event.code === 'Semicolon') {
+      if (this.allowAnyEmail) {
+        event.preventDefault();
+        event.stopPropagation();
+      } else {
+        console.warn(strings.anyEmailWarning);
+      }
+      return;
     } else {
       this.userInput = input.value;
-      this.handleUserSearch();
+      const validEmail = isValidEmail(this.userInput);
+      if (!validEmail) {
+        this.handleUserSearch();
+      }
+    }
+  }
+
+  private handleAnyEmail() {
+    this._showLoading = false;
+    this._arrowSelectionCount = 0;
+    if (isValidEmail(this.userInput)) {
+      const anyMailUser = {
+        mail: this.userInput,
+        displayName: this.userInput,
+        id: this.userInput
```

Comment:
It can be avoided. There are a few methods using the id property like the `addPerson` and `removePerson` ones but those can be updated to cater for lack of ids. I'm checking the rest of the other instances where ids are used if they would require this update.